src/V4Router.sol

```diff
@@ -48,7 +50,7 @@ abstract contract V4Router is IV4Router, BaseActionsRouter, DeltaResolver {
                 _settle(currency, _msgSender(), _getFullSettleAmount(currency));
             } else if (action == Actions.SETTLE_WITH_BALANCE) {
                 Currency currency = params.decodeCurrency();
-                _settle(currency, address(this), _getFullSettleAmount(currency));
+                _settle(currency, address(this), currency.balanceOfSelf());
```

Comment:
this change is specifically for FOT support yes? we need this then in posm too..

Reply:
just noting that the behavior for this command is changing here and i want to keep the same commands that are needed in posm and routing behaving the same.

Im also confused are we still adding a command settle that takes in an amount?? or take with an amount?? 

ie in final we will have?

settle(curreny,amount) -> amount can be the constant OPEN_DELTA which means query it
settle_all -> current behavior
settleWithBalance(currency) -> amount is derived from the contract balance and payer is address(this)

Reply:
discussed on slack but this is not (solely) for FOT although it doesnt make a sub-optimal FOT option possible. This is to enable multi-hop multi-protocol swaps like
USDC-v2->DAI-v4->UNI
where the DAI gets left in the contract and then passed in to the second swap using SETTLE_WITH_BALANCE and OPEN_DELTA

separately i will add a settle that takes an amount for FOT to be optimal